core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -383,8 +383,8 @@ private[spark] class DAGScheduler(
   /**
    * Called by the TaskSetManager when it decides a speculative task is needed.
    */
-  def speculativeTaskSubmitted(task: Task[_]): Unit = {
-    eventProcessLoop.post(SpeculativeTaskSubmitted(task))
+  def speculativeTaskSubmitted(task: Task[_], taskIndex: Int = -1): Unit = {
```

Comment:
We dont need the default value.

```suggestion
  def speculativeTaskSubmitted(task: Task[_], taskIndex: Int): Unit = {
```